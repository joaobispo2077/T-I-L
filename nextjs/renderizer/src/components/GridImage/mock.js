export default {
	hasBackground: false,
	title: 'My grid',
	description:
		'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
	grid: [
		{
			altText: 'Teste 1',
			src: 'https://source.unsplash.com/random/800x800?r=1',
		},
		{
			altText: 'Teste 2',
			src: 'https://source.unsplash.com/random/800x800?r=2',
		},
		{
			altText: 'Teste 3',
			src: 'https://source.unsplash.com/random/800x800?r=3',
		},
		{
			altText: 'Teste 4',
			src: 'https://source.unsplash.com/random/800x800?r=4',
		},
		{
			altText: 'Teste 5',
			src: 'https://source.unsplash.com/random/800x800?r=5',
		},
		{
			altText: 'Teste 6',
			src: 'https://source.unsplash.com/random/800x800?r=6',
		},
	],
};
