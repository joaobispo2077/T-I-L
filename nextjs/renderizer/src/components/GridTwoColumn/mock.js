export default {
  title: 'GridTwoColumn',
  description:
    'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Suscipit quisquam aspernatur sed quos, dolore cum eveniet magnam at consequatur totam iste minus, blanditiis obcaecati eos rerum sunt dolorum perferendis. Nam?',
  image: {
    url: 'assets/images/javascript.svg',
    alternativeText: 'bla bla',
  },
  background: true,
};
