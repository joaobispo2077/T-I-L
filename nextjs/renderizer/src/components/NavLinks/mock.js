export default [
	{
		id: 1,
		text: 'Link 1',
		link: '#target1',
		newTab: false,
	},
	{
		id: 2,
		text: 'Link 2',
		link: '#target2',
		newTab: false,
	},
	{
		id: 3,
		text: 'Link 3',
		link: '#target3',
		newTab: true,
	},
	{
		id: 4,
		text: 'Link 4',
		link: '#target4',
		newTab: false,
	},
	{
		id: 5,
		text: 'Link 5',
		link: '#target5',
		newTab: false,
	},
];
