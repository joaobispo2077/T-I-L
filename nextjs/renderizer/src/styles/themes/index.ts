export const themes = {
  default: {
    colors: {
      primary: '#0A1128',
      secondary: '#dc143C',
      light: '#FFFFFF',
      dark: '#0A1128',
      mediumGray: '#DDDDDD',
    },
    font: {
      family: {
        primary: "'Open Sans', sans-serif",
        secondary: "'Montserrat', sans-serif",
      },
      sizes: {
        xsmall: '.8rem',
        small: '1.6rem',
        medium: '2.4rem',
        large: '3.2rem',
        xlarge: '4.0rem',
        xxlarge: '4.8rem',
        huge: '5.6rem',
        xhuge: '6.4rem',
      },
    },
    spacings: {
      xsmall: '.8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
    },
    media: {
      lteMedium: '(max-width: 768px)',
    },
  },
} as const;
