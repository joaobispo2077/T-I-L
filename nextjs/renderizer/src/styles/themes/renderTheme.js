import { ThemeProvider } from 'styled-components';
import { themes } from '.';

export const renderTheme = (children) => {
	return (
		<ThemeProvider theme={{ ...themes['default'] }}>{children}</ThemeProvider>
	);
};
