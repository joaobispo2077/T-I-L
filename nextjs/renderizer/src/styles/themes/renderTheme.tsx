import React from 'react';
import { ThemeProvider } from 'styled-components';
import { themes } from '.';

export const renderTheme = (children: React.ReactNode) => {
  return (
    <ThemeProvider theme={{ ...themes.default }}>{children}</ThemeProvider>
  );
};
