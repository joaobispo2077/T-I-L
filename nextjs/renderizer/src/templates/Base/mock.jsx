import linksMock from '../../components/NavLinks/mock';

import { GridText } from '../../components/GridText';
import gridMock from '../../components/GridText/mock';

export default {
	links: linksMock,
	logo: {
		text: 'Logo',
		link: '#',
	},
	footer: '<p>testando footer</p>',
	children: (
		<div>
			<GridText {...gridMock} hasBackground />
			<GridText {...gridMock} />
			<GridText {...gridMock} hasBackground />
			<GridText {...gridMock} />
		</div>
	),
};
