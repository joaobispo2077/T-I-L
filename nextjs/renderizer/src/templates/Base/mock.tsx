import linksMock from '../../components/NavLinks/mock';

import { GridText } from '../../components/GridText';
import gridMock from '../../components/GridText/mock';
import { BaseProps } from '.';

export default {
  links: linksMock,
  logo: {
    text: 'Logo',
    link: '#',
  },
  footer: '<p>testando footer</p>',
  children: (
    <div>
      <GridText {...gridMock} hasBackground />
      <GridText {...gridMock} />
      <GridText {...gridMock} hasBackground />
      <GridText {...gridMock} />
    </div>
  ),
} as BaseProps;
