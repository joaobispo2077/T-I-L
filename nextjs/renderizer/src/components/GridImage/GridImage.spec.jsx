import { render } from '@testing-library/react';
import { GridImage } from '.';
import { renderTheme } from '../../styles/themes/renderTheme';
import mock from './mock';

describe('<GridImage/>', () => {
	it('should match with snapshot', () => {
		const { container } = render(renderTheme(<GridImage {...mock} />));

		expect(container).toMatchSnapshot();
	});
});
