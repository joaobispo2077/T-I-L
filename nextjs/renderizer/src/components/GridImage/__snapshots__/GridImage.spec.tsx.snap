// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridImage/> should match with snapshot 1`] = `
.c5 {
  font-size: 2.4rem;
}

.c2 .c4 {
  margin-bottom: 3.2rem;
}

.c6 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 3.2rem;
}

.c7 {
  overflow: hidden;
}

.c8 {
  width: 100%;
  cursor: pointer;
  -webkit-transition: -webkit-transform 0.25s ease-in-out;
  -webkit-transition: transform 0.25s ease-in-out;
  transition: transform 0.25s ease-in-out;
}

.c8:hover {
  -webkit-transform: scale(1.2) rotate(10deg);
  -ms-transform: scale(1.2) rotate(10deg);
  transform: scale(1.2) rotate(10deg);
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

@media (max-width:768px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

<div>
  <section
    class="c0"
    id=""
  >
    <section
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          My grid
        </h2>
        <footer
          class="c4 c5"
        >
          Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
        </footer>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="Teste 1"
              class="c8"
              src="https://source.unsplash.com/random/800x800?r=1"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 2"
              class="c8"
              src="https://source.unsplash.com/random/800x800?r=2"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 3"
              class="c8"
              src="https://source.unsplash.com/random/800x800?r=3"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 4"
              class="c8"
              src="https://source.unsplash.com/random/800x800?r=4"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 5"
              class="c8"
              src="https://source.unsplash.com/random/800x800?r=5"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Teste 6"
              class="c8"
              src="https://source.unsplash.com/random/800x800?r=6"
            />
          </div>
        </div>
      </div>
    </section>
  </section>
</div>
`;
